core/src/main/java/org/apache/iceberg/avro/AvroFormatModel.java

```diff
@@ -0,0 +1,272 @@
+
+    @Override
+    public CloseableIterable<D> build() {
+      return internal
+          .createResolvingReader(unused -> readerFunction.apply(icebergSchema, idToConstant))
+          .build();
+    }
+  }
+
+  private static class WriteBuilderWrapper<D, S> implements WriteBuilder<D, S> {
+    private final Avro.WriteBuilder internal;
+    private final BiFunction<org.apache.avro.Schema, S, DatumWriter<D>> writerFunction;
+    private S inputSchema;
+
+    private WriteBuilderWrapper(
+        EncryptedOutputFile outputFile,
+        BiFunction<org.apache.avro.Schema, S, DatumWriter<D>> writerFunction) {
+      this.internal = Avro.write(outputFile.encryptingOutputFile());
+      this.writerFunction = writerFunction;
+    }
+
+    @Override
+    public WriteBuilder<D, S> schema(Schema schema) {
+      internal.schema(schema);
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> inputSchema(S schema) {
+      this.inputSchema = schema;
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> set(String property, String value) {
+      internal.set(property, value);
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> setAll(Map<String, String> properties) {
+      internal.setAll(properties);
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> meta(String property, String value) {
+      internal.meta(property, value);
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> meta(Map<String, String> properties) {
+      internal.meta(properties);
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> content(FileContent content) {
+      switch (content) {
+        case DATA:
+          internal.createContextFunc(Avro.WriteBuilder.Context::dataContext);
+          internal.createWriterFunc(avroSchema -> writerFunction.apply(avroSchema, inputSchema));
+          break;
+        case EQUALITY_DELETES:
+          internal.createContextFunc(Avro.WriteBuilder.Context::deleteContext);
+          internal.createWriterFunc(avroSchema -> writerFunction.apply(avroSchema, inputSchema));
+          break;
+        case POSITION_DELETES:
+          internal.createContextFunc(Avro.WriteBuilder.Context::deleteContext);
+          internal.createWriterFunc(unused -> new Avro.PositionDatumWriter());
+          internal.schema(DeleteSchemaUtil.pathPosSchema());
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown file content: " + content);
+      }
+
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> metricsConfig(MetricsConfig metricsConfig) {
+      internal.metricsConfig(metricsConfig);
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> overwrite() {
+      internal.overwrite();
+      return this;
+    }
+
+    @Override
+    public WriteBuilder<D, S> withFileEncryptionKey(ByteBuffer encryptionKey) {
+      // Avro doesn't support file encryption
```

Comment:
Nit: This comment duplicates the information on the next line so it isn't adding any value.

Reply:
Removed